include/astshim/ChebyMap.h

```diff
@@ -0,0 +1,321 @@
+    @throw std::invalid_argument if the forward transform is specified (coeff_f is not empty)
+        and lbnd_f or ubnd_f do not have nin elements.
+    @throw std::invalid_argument if the inverse transform is specified (coeff_i is not empty)
+        and lbnd_i or ubnd_i do not have nout elements.
+
+    @anchor ChebyMap_CoefficientMatrices Coefficient Matrices
+
+    The coefficients describing a forward transformation are specified as an `ncoeff_f x (2 + nin)`
+    matrix, where `ncoeff_f` is the number of coefficients for the forward
+    direction, and each coefficient is described by `2 + nin` contiguous values, as follows:
+    - The first element is the coefficient value.
+    - The next element is the integer index of the @ref ChebyMap output
+        which uses the coefficient within its defining polynomial
+        (the first output has index 1).
+    - The remaining elements give the Chebyshev order to use with each corresponding input coordinate
+        value, or 0 to ignore that input coordinate. Powers must not be negative and floating point
+        values are rounded to the nearest integer.
+
+    For instance, if the @ref ChebyMap has 3 inputs and 2 outputs, each row consisting
+    of 5 elements, and the row `(1.2, 2, 6, 3, 0)` describes a coefficient
+    that increments output 2 as follows:
+        `out2 += 1.2 * T6(in1') * T3(in2') * T0(in3')
+    and the row `(-1.5, 1, 0, 0, 0)` describes a coefficient that increments
+    output 2 with a constant value of -1.5 (since all powers are 0):
+        `out1 += -1.5 * T0(in1') * T0(in2') * T0(in3')
+    where inI' is the normalized value of input axis I
+    Each final output coordinate value is the sum of the terms described
+    by the `ncoeff_f` columns in the supplied array. If no coefficients increment
+    a given output then it will be set to 0.
+
+    The coefficients describing the inverse transformation work the same way,
+    but of course the matrix is of size `ncoeff_i x (2 + nout)`, where `ncoeff_i` is the
+    number of coefficients for the inverse transformation.
+    */
+    explicit ChebyMap(ndarray::Array<double, 2, 2> const &coeff_f,
+                      ndarray::Array<double, 2, 2> const &coeff_i, std::vector<double> const &lbnd_f,
+                      std::vector<double> const &ubnd_f, std::vector<double> const &lbnd_i,
+                      std::vector<double> const &ubnd_i, std::string const &options = "")
+            : Mapping(reinterpret_cast<AstMapping *>(
+                      _makeRawChebyMap(coeff_f, coeff_i, lbnd_f, ubnd_f, lbnd_i, ubnd_i, options))) {}
+
+    /**
+    Construct a @ref ChebyMap with only the forward transform specified.
+
+    If the polynomial is invertible and you want an inverse can you call polyTran to fit one
+    (at this time the iterative inverse offered by PolyMap is not available for ChebyMap).
+
+    @param[in] coeff_f  A `(2 + nin) x ncoeff_f` @ref ChebyMap_CoefficientMatrices "matrix of coefficients"
```

Comment:
I could, but Doxygen will complain if I only document one parameter, and there are subtle differences between the two methods, such as the fact that `coeff_f` must be specified.